File: python_pkg/python_pkg/robot_station.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotStation(Node):
    def __init__(self):
        super().__init__("robot_station")

        self.declare_parameter("robot_name", "EJSC")
        self.robot_name = self.get_parameter("robot_name").value
        
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.timer_ = self.create_timer(0.5, self.publisher_news)
        self.get_logger().info("\033[32mRobot Station News has been started\033[m")


    def publisher_news(self):

        msg = String()
        msg.data = f"\033[33mTá saindo da jaula o monstro {self.robot_name}\033[m"
        self.publisher_.publish(msg)


def main(args=None):
    
    rclpy.init(args=args)

    node = RobotStation()

    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: cpp_packages/cpp_pkg/src/robot_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotStation: public rclcpp::Node{
    private:
        rclcpp::TimerBase::SharedPtr timer_;
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        
        void publisher_news(){
            example_interfaces::msg::String msg;
            msg.data = std::string("Tá saindo da jaula o monstro ") + robot_name;
            publisher_->publish(msg);
        }

        std::string robot_name;

    public:

        RobotStation():Node("robot_station"){
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
            this->declare_parameter("robot_name", "EJSC");
            robot_name = this->get_parameter("robot_name").as_string();
            timer_ = this->create_wall_timer(std::chrono::milliseconds(500), 
                                             std::bind(&RobotStation::publisher_news, this));
            RCLCPP_INFO(this->get_logger(), "Robot Station News has been started");
        }

};

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotStation>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}